components/server/src/server.rs

```diff
@@ -1953,25 +1954,39 @@ impl<T: fmt::Display + Send + 'static> Stop for LazyWorker<T> {
     }
 }
 
-pub struct EngineMetricsManager<EK: KvEngine, R: RaftEngine> {
-    engines: Engines<EK, R>,
+pub struct EngineMetricsManager<EK: KvEngine, ER: RaftEngine> {
+    tablet_registry: TabletRegistry<EK>,
+    raft_engine: ER,
     last_reset: Instant,
 }
 
-impl<EK: KvEngine, R: RaftEngine> EngineMetricsManager<EK, R> {
-    pub fn new(engines: Engines<EK, R>) -> Self {
+impl<EK: KvEngine, ER: RaftEngine> EngineMetricsManager<EK, ER> {
+    pub fn new(tablet_registry: TabletRegistry<EK>, raft_engine: ER) -> Self {
         EngineMetricsManager {
-            engines,
+            tablet_registry,
+            raft_engine,
             last_reset: Instant::now(),
         }
     }
 
     pub fn flush(&mut self, now: Instant) {
-        KvEngine::flush_metrics(&self.engines.kv, "kv");
-        self.engines.raft.flush_metrics("raft");
-        if now.saturating_duration_since(self.last_reset) >= DEFAULT_ENGINE_METRICS_RESET_INTERVAL {
-            KvEngine::reset_statistics(&self.engines.kv);
-            self.engines.raft.reset_statistics();
+        let should_reset =
+            now.saturating_duration_since(self.last_reset) >= DEFAULT_ENGINE_METRICS_RESET_INTERVAL;
+        let mut is_first_instance = true;
+        self.tablet_registry
+            .for_each_opened_tablet(&mut |_, tablet: &mut CachedTablet<EK>| {
+                if let Some(tablet) = tablet.latest() {
```

Comment:
If you can make the method defined on statistics, then you don't need to care whether it's shared, just `registry.statistics.flush(); for_each(|t| t.flush_other());`.